src/Castle.Core/DynamicProxy/Generators/Emitters/SimpleAST/DefaultValueExpression.cs

```diff
@@ -72,7 +72,7 @@ private void EmitByRef(ILGenerator gen)
 
 		private bool IsPrimitiveOrClass(Type type)
 		{
-			if ((type.IsPrimitive && type != typeof(IntPtr)))
+			if (type.IsPrimitive && type != typeof(IntPtr) && type != typeof(UIntPtr))
```

Comment:
Maybe worth noting that this double check for both `IntPtr` and `UIntPtr` is already present in other locations; it was only missing here.